File: Contest 44 Quarter #2/Q3 1465. 任务顺序/li.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    /**
     * @param n: The number of tasks
     * @param t: The time array t
     * @param p: The probability array p
     * @return: Return the order
     */
    vector<int> getOrder(int n, vector<int> t, vector<double> p) {
        vector<pair<double, int>> v;
        for (int i = 0; i < n; ++i)
            v.push_back({ -(double)p[i] / t[i], i + 1});
        sort(v.begin(), v.end());
        vector<int> res;
        for (int i = 0; i < n; ++i)
            res.push_back(v[i].second);
        return res;
    }
};

/********************* test case *********************/
int main() {
    Solution s;
    // for (int i : s.getOrder(2, {1, 2}, {0.3, 0.7}))
    //     cout << i << " ";
    double a = 2/0.2;
    double b = 0.7/7;
    double c = 0.1;
    cout << a << endl;
    cout << b << endl;
    cout << (a == b) << endl;
    cout << (a == c) << endl;
    cout << (b == c) << endl;
}

File: Contest 29 Weekly #16/1396.Set-Union/li.py
class Solution:
    """
    @param sets: Initial set list
    @return: The final number of sets
    """
    def setUnion(self, sets):
        parent = {}
        for s in sets:
            for i in s:
                parent[i] = i
        def find(x):
            if parent[x] == x:
                return x
            parent[x] = find(parent[x])
            return parent[x]
        def union(x, y):
            x, y = find(x), find(y)
            parent[x] = y
            return x != y

        for s in sets:
            if not s:
                continue
            x = s[0]
            for y in s[1:]:
                union(x, y)
        res = 0
        for i in parent:
            if parent[i] == i:
                res += 1
        # print parent
        return res


############ test case ###########
s = Solution()
print s.setUnion([[1, 2, 3], [3, 9, 7], [4, 5, 10]])
